packages/fuels/src/cli/types.ts

```diff
@@ -0,0 +1,79 @@
+import type { DeployContractOptions } from '@fuel-ts/contract';
+
+export enum Commands {
+  build = 'build',
+  deploy = 'deploy',
+  dev = 'dev',
+  init = 'init',
+}
+
+export type CommandEvent =
+  | {
+      type: Commands.build;
+      data: unknown;
+    }
+  | {
+      type: Commands.deploy;
+      data: ContractDeployed[];
+    }
+  | {
+      type: Commands.dev;
+      data: unknown;
+    }
+  | {
+      type: Commands.init;
+      data: unknown;
+    };
+
+export type ContractDeployed = {
+  name: string;
+  contractId: string;
+};
+
+export type DeployOptions = {
+  contracts: ContractDeployed[];
+  contractName: string;
+  contractPath: string;
+};
+
+export type OptionsFunction = (
+  options: DeployOptions
+) => DeployContractOptions | Promise<DeployContractOptions>;
+
+export type UserFuelsConfig = {
+  workspace?: string;
+  contracts?: string[];
+  predicates?: string[];
+  scripts?: string[];
+  output: string;
+
+  privateKey?: string;
+  providerUrl?: string;
```

Comment:
`providerUrl` and `fuelCorePort` seem like conflicting properties, e.g.:

```ts
const config = {
    providerUrl: "http://127.0.0.1:4000/graphql",
    fuelCorePort: 4001
}
```

Reply:
I added better docs to clarify this. https://github.com/FuelLabs/fuels-ts/pull/1291/commits/40ec0117ba74b4a5afb586acc099296aeb5cb755

----

The `fuelCorePort` property is only used by the `fuels dev` command when `autoStartFuelCore` is set to `true`.

During the `fuels dev` command, the `providerUrl` is overridden with that of the newly started `fuel-core` node.

For things like `fuels deploy`, the config `providerUrl` is used, and `fuelCorePort` is ignored.